consensus/misc/kip71_test.go

```diff
@@ -0,0 +1,140 @@
+package misc
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/params"
+)
+
+func getTestConfig(forkedBlockNum *big.Int) *params.ChainConfig {
+	testConfig := params.CypressChainConfig
+	testConfig.UnitPrice = uint64(25000000000)
+	testConfig.KIP71CompatibleBlock = forkedBlockNum
+	testConfig.Governance = &params.GovernanceConfig{
+		KIP71: params.GetDefaultKip71Config(),
+	}
+	return testConfig
+}
+
+func TestCalcBaseFee(t *testing.T) {
+	tests := []struct {
+		parentBaseFee int64
+		parentGasUsed uint64
+		nextBaseFee   int64
+	}{
+		{750000000000, 30000000, 750000000000}, // usage == target
+		{30000000000, 20000000, 29500000000},   // usage below target
+		{300000000000, 40000000, 305000000000}, // usage above target
+	}
+	for i, test := range tests {
+		parent := &types.Header{
+			Number:  common.Big3,
+			GasUsed: test.parentGasUsed,
+			BaseFee: big.NewInt(test.parentBaseFee),
+		}
+		if have, want := CalcBaseFee(parent, getTestConfig(big.NewInt(3))), big.NewInt(test.nextBaseFee); have.Cmp(want) != 0 {
+			t.Errorf("test %d: have %d  want %d, ", i, have, want)
+		}
+	}
+}
+
+func TestBlockNumReachDoubleBaseFee(t *testing.T) {
+	blockNum := 0
+	parentBaseFee := big.NewInt(25000000000)
+	for i := 0; i < 15; i++ {
+		parent := &types.Header{
+			Number:  common.Big3,
+			GasUsed: 84000000,
+			BaseFee: parentBaseFee,
+		}
+		parentBaseFee = CalcBaseFee(parent, getTestConfig(big.NewInt(3)))
+		blockNum = i
+		// t.Logf("test %d: have %d, ", i, parentBaseFee)
+	}
+	if parentBaseFee.Cmp(big.NewInt(25000000000*2)) < 0 {
+		t.Errorf("block number %d: have %d want < %d", blockNum, parentBaseFee, 25000000000*2)
+	}
+}
```

Comment:
Yes, so we need to change the `garTarget` carefully.
We can process the case, `parentGasUsed` - `gasTarget` < 20, as infinitesimal because we treat a large number like ston, klay

Reply:
> Yes, so we need to change the garTarget carefully.

How do you expect it? How do you anticipate GCs are aware of this doubled condition? Or, can you propose an exact equation for the doubled condition?

> We can process the case, parentGasUsed - gasTarget < 20, as infinitesimal because we treat a large number like ston, klay

Could you elaborate on this?

Reply:
1. I think we don't need exact equation now. As long as it's reasonable enough to control malicious bulk transactions while ensuring network usability, that's fine. If you cannot agree, could you explain the problematic situation?
2. Usually gas consumption is 10 million units so we can treat a value < 20 as infinitesimal.

Reply:
1. My originally intended question was to ask whether the test semantic is correct, as the doubled condition is variable by `gasTarget`, `parentBaseFee`, and `denominator`. The doubled condition in 15 blocks seems not still preserved even If the `parentBaseFee` is the max value.
2. Still confused. The smaller of `parentGasUsed - gasTarget` seems not to make doubled `baseFee`. The calculated `baseFee` slowly increases as that difference is small as possible. What do you mean `parentGasUsed - gasTarget < 20`?

Reply:
1. I understood what you mean. However please don't think of it as a general and perfect test case, this test case is just a test to roughly confirm that our intended behavior is working with default parameters. If the parameters are adjusted, the testcases should be done also.
2. Doubling the `baseFee` is intended only for peak usage

Reply:
I think you are looking at the previous version. Please click the `files changed`, or check the refactored commit([3fcd8e5](https://github.com/klaytn/klaytn/pull/1409/commits/3fcd8e5f503b17cca1cf97efee90957f9ef77b73)).

Reply:
1. If you want to hold that test case, please add enough comments to avoid confusion. Even if I feel your intention, the function as it creates confusion for developers because the semantic is not correct. 
2. The doubling is done via continuous increasing of `baseFee`. Why do you think the doubling is created only with peak usage?  The doubling can be created within 15 blocks although the parent's `baseFee` is not upperbound(`84000000`),

I was originally asking what does your comment means?
> Usually gas consumption is 10 million units so we can treat a value < 20 as infinitesimal.

Okay: `Usually gas consumption is 10 million units` 
Confused:  `so we can treat a value < 20 as infinitesimal`
What does that mean `so we can treat a value < 20 as infinitesimal`?